File: backend/src/main/java/com/horain/repository/AgentTurnRepository.java
package com.horain.repository;

import com.horain.model.AgentTurn;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

/**
 * JPA repository for agent conversation turns.
 */
public interface AgentTurnRepository extends JpaRepository<AgentTurn, UUID> {

    List<AgentTurn> findByConversationIdOrderByTurnIndexAsc(UUID conversationId);

    List<AgentTurn> findByCreatedAtAfterOrderByCreatedAtDesc(Instant after, org.springframework.data.domain.Pageable pageable);

    List<AgentTurn> findByStatusIn(List<String> statuses);
}

File: backend/src/main/kotlin/com/horain/repository/AgentTurnRepository.kt
package com.horain.repository

import com.horain.model.AgentTurn
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.time.Instant
import java.util.UUID

/**
 * JPA repository for agent conversation turns.
 */
interface AgentTurnRepository : JpaRepository<AgentTurn, UUID> {

    fun countByConversationId(conversationId: UUID): Long

    fun findByConversationIdOrderByTurnIndexAsc(conversationId: UUID): List<AgentTurn>

    fun findByCreatedAtAfterOrderByCreatedAtDesc(after: Instant, pageable: Pageable): List<AgentTurn>

    fun findByStatusIn(statuses: List<String>): List<AgentTurn>
}
